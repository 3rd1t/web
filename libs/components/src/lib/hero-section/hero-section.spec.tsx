import React from "react"
import renderer from "react-test-renderer"
import HeroSection from "./hero-section"

describe(" HeroSection", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <HeroSection
          color="primary"
          headline={{
            regular: "Enhance your research with",
            highlight: "factor returns",
          }}
          paragraph="Factor investing is a strategy that chooses securities on attributes that are associated with higher returns. There are two main types of factors that have driven returns of stocks, bonds, and other factors: macroeconomic factors and style factors. The former captures broad risks across asset classes while the latter aims to explain returns and risks within asset classes."
          primaryButton={{
            label: "Get started",
            href: "#",
          }}
          secondaryButton={{
            label: "Learn more",
            href: "#",
          }}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
