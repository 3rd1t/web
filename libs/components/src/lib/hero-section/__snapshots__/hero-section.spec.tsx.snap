// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` HeroSection renders correctly 1`] = `
<div
  className="flex flex-col items-center w-full"
>
  <div
    className="space-y-4 text-left sm:text-center md:space-y-8 xl:space-y-12 lg:text-left"
  >
    <h2
      className="text-4xl font-extrabold leading-10 tracking-tight text-gray-900 xl:text-5xl sm:leading-none md:text-5xl"
    >
      Enhance your research with
      <br />
      <span
        className="text-primary-600"
      >
         
        factor returns
      </span>
    </h2>
    <p
      className="text-gray-700 sm:text-lg sm:mx-auto md:text-xl lg:mx-0"
    >
      Factor investing is a strategy that chooses securities on attributes that are associated with higher returns. There are two main types of factors that have driven returns of stocks, bonds, and other factors: macroeconomic factors and style factors. The former captures broad risks across asset classes while the latter aims to explain returns and risks within asset classes.
    </p>
    <div
      className="space-y-4 md:space-x-4 md:space-y-0"
    >
      <div
        className="sm:flex sm:justify-center lg:justify-start"
      >
        <div
          className="rounded-md shadow"
        >
          <a
            className="border border-transparent rounded-md md:py-4 md:text-lg md:px-10 flex items-center justify-center w-full px-8 py-3 text-base font-medium leading-6 transition duration-150 ease-in-out focus:outline-none text-white bg-primary-600 hover:bg-primary-400 hover:text-primary-100"
            href="#"
          >
            Get started
          </a>
        </div>
        <div
          className="mt-3 sm:mt-0 sm:ml-3"
        >
          <a
            className="border border-transparent rounded-md md:py-4 md:text-lg md:px-10 flex items-center justify-center w-full px-8 py-3 text-base font-medium leading-6 transition duration-150 ease-in-out focus:outline-none text-primary-700 bg-primary-100 hover:text-primary-600 hover:bg-white"
            href="Learn more"
          >
            Learn more
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
